contracts/ChipRegistry.sol

```diff
@@ -417,267 +377,100 @@ contract ChipRegistry is IChipRegistry, ClaimedPBT, Ownable {
+     * @return                  The owner of the chip
      */
-    function tokenURI(address _chipId) public view override returns (string memory) {
-        string memory tokenUri = _getChipPrimaryServiceContentByRecordType(_chipId, URI_RECORDTYPE);
-        return bytes(tokenUri).length == 0 ? ClaimedPBT.tokenURI(_chipId) : tokenUri;
-    }
+    function ownerOf(address _chipId) public view virtual returns (address) {
+        require(chipEnrollments[_chipId].chipEnrolled, "Chip not added");
+        IPBT projectRegistrar = IPBT(chipEnrollments[_chipId].projectRegistrar);
 
-    function getGatewayUrls() external view returns(string[] memory) {
-        return gatewayUrls;
-    }
+        return projectRegistrar.ownerOf(_chipId);
+    } 
 
     /* ============ Internal Functions ============ */
 
     /**
-     * Get ERS node from tokenData and then sets the new Owner of the chip on the ERSRegistry.
-     */
-    function _setERSOwnerForChip(address _chipId, address _newOwner) internal {
-        (bytes32 chipErsNode, ) = _decodeTokenData(chipTable[_chipId].tokenData);
-        ers.setNodeOwner(chipErsNode, _newOwner);
-    }
-
-    /**
-     * Check that certificates passed as part of claim are valid. Developer cert is valid if the project public key signed
-     * the address of the chip. We then check the validity of the signed certificate which is the project public key
-     * signed by the chip.
+     * @notice Create a chip subnode record in the ERS; if resolver is not set, set to services registry
+     *
+     * @param _rootNode         The root node of the project
+     * @param _nameHash         The namehash of the chip
+     * @param _owner            The owner of the chip
+     * @param _resolver         The resolver of the chip
+     * @return                  The ERS node of the chip
      */
-    function _validateCertificates(
-        address _chipId,
-        address _projectPublicKey,
-        bytes memory _developerInclusionProof,
-        bytes memory _developerCustodyProof
+    function _createChipSubnode(
+        bytes32 _rootNode,
+        bytes32 _nameHash,
+        address _owner,
+        address _resolver
     )
         internal
-        view
+        returns (bytes32)
     {
-        (bool validCertificates, string memory errorMessage) = _areValidCertificates(
-            _chipId,
-            _projectPublicKey,
-            _developerInclusionProof,
-            _developerCustodyProof
+        // Create the chip subnode record in the ERS; if the node already exists, this should revert
+        ers.createChipRegistrySubnodeRecord(
+            _rootNode, 
+            _nameHash, 
+            _owner, 
+            _resolver
         );
 
-        require(validCertificates, errorMessage);
+        // Verify the chip's ERS node was created by the ProjectRegistrar; this is the source of truth for the chip's ownership
+        return keccak256(abi.encodePacked(_rootNode, _nameHash));
```

Comment:
Believe this value is returned by `ers.createChipRegistrySubnodeRecord`

Reply:
Added in WIP commit.